pydantic_ai_slim/pydantic_ai/toolsets/skills/_toolset.py

```diff
@@ -0,0 +1,411 @@
+            # Find the script
+            script = None
+            for s in skill.scripts:
+                if s.name == script_name:
+                    script = s
+                    break
+
+            if script is None:
+                available = [s.name for s in skill.scripts]
+                return (
+                    f"Error: Script '{script_name}' not found in skill '{skill_name}'. Available scripts: {available}"
+                )
+
+            # Security check
+            if not _is_safe_path(skill.path, script.path):
+                return 'Error: Script path escapes skill directory.'
+
+            # Build command
+            cmd = [self._python_executable, str(script.path)]
+            if args:
+                cmd.extend(args)
+
+            try:
+                # Use anyio.run_process for async-compatible execution
+                result = None
+                with anyio.move_on_after(self._script_timeout) as scope:
+                    result = await anyio.run_process(
+                        cmd,
+                        check=False,  # We handle return codes manually
+                        cwd=str(skill.path),
+                    )
+
+                # Check if timeout was reached
+                if scope.cancelled_caught:
+                    raise SkillScriptExecutionError(
+                        f"Script '{script_name}' timed out after {self._script_timeout} seconds"
+                    )
+
+                # At this point, result should be set (timeout check passed)
+                assert result is not None
+
+                # Decode output from bytes to string
+                output = result.stdout.decode('utf-8', errors='replace')
+                if result.stderr:
+                    stderr = result.stderr.decode('utf-8', errors='replace')
+                    output += f'\n\nStderr:\n{stderr}'
+
+                if result.returncode != 0:
+                    output += f'\n\nScript exited with code {result.returncode}'
+
+                return output.strip() or '(no output)'
+
+            except OSError as e:
+                raise SkillScriptExecutionError(f"Failed to execute script '{script_name}': {e}") from e
+
+    def get_skills_system_prompt(self) -> str:
+        """Get the combined system prompt from all loaded skills.
+
+        This should be added to the agent's system prompt to provide
+        skill discovery and usage instructions.
+
+        Following Anthropic's approach, this includes all skill metadata upfront
+        in the system prompt, enabling the agent to discover and select skills
+        without needing to call list_skills() first.
+
+        Returns:
+            Formatted system prompt containing:
+            - All skill metadata (name + description)
+            - Instructions for using skill tools
+            - Progressive disclosure guidance
+        """
+        if not self._skills:
+            return ''
+
+        lines = [
+            '# Skills',
```

Comment:
Default prompts should live in constants at the top of the file and should be overridable, either through the constructor, or at least the in-development https://github.com/pydantic/pydantic-ai/pull/3656